.circleci/config.yml

```diff
@@ -115,12 +135,171 @@ jobs:
           command: |
             cd languages/java/app-encryption
             ./scripts/test.sh
+      - store_test_results:
+          path: languages/java/app-encryption/target/surefire-reports
       - save_cache:
           paths:
             - ~/.m2
-          key: v1-maven-{{ .Branch }}-{{ checksum "languages/java/secure-memory/pom.xml" }}
-      - store_test_results:
-          path: languages/java/app-encryption/target/surefire-reports
+          key: v1-maven-{{ .Branch }}-{{ checksum "/tmp/checksum.txt" }}
       - codecov/upload:
           file: languages/java/app-encryption/target/site/jacoco/jacoco.xml
           flags: languages_java_app_encryption
+  languages-csharp-logging:
+    docker:
+      - image: mcr.microsoft.com/dotnet/core/sdk:2.2
+    steps:
+      - checkout
+      - run:
+          name: Generate cache key
+          command: |
+            cd languages/csharp/Logging
+            ../../../scripts/checksum.sh /tmp/checksum.txt *.csproj
+      - restore_cache:
+          keys:
+            # when project files change, use increasingly general patterns to restore cache.
+            # vN prefix in case we ever need to regenerate all caches
+            - v1-nuget-{{ .Branch }}-{{ checksum "/tmp/checksum.txt" }}
+            - v1-nuget-{{ .Branch }}-
+            - v1-nuget-
+      - run:
+          name: Build
+          command: |
+            cd languages/csharp/Logging
+            ./scripts/clean.sh
+            ./scripts/build.sh
+      - run:
+          name: Tests
+          command: |
+            cd languages/csharp/Logging
+            ./scripts/test.sh
+      - run:
+          name: Convert test results
+          command: |
+            cd languages/csharp/Logging
+            dotnet tool install -g trx2junit
+            export PATH="$PATH:/root/.dotnet/tools"
+            trx2junit Logging.Tests/TestResults/*.trx
+      - store_test_results:
+          path: languages/csharp/Logging/Logging.Tests/TestResults
+      - save_cache:
```

Comment:
i think this should be moved up to run sooner (once we know all possible dependencies have been pulled down). the Tests (test.sh) step should theoretically be the last possible place they could be pulled down.

this also has an added benefit of the convert test results, store_test_results and codecov steps all being logically grouped together